### DISCUSSION TITLE
Quando se originou o desenvolvedor Front End?

### DISCUSSION THREAD

**Discussion by vitocaio** [frontendbr/forum#707] [Discussão]
Boa tarde meus caros, em uma discussão com um colega de trabalho, nos deparamos com a seguinte questão...
Quando se originou o desenvolvedor Front End?
Para mim foi no surgimento do Jquery em meados de 2007, foi ali que as skills para receber tal titulo surgiu e para vocês?

**Comment by rubenmarcus** [comment]
Olha, pra mim foi com o surgimento do Node e Frameworks SPAS. Eu sou front (empregado e profissional) desde 2011, mas até meados de 2013 eu não sabia que era front. Pra mim era um misto de webdesigner / webmaster.

Quando JS começou a ganhar mais notoriedade na comunidade (pelo menos pra mim) Foi que Front acabou ganhando relevância.

**Comment by bernardodiasc** [comment]
Acho que coincide mais com a época do surgimento de dispositivos móveis com telas e com recursos de browser engines. Foi quando alguns profissionais começaram a se engajar mais em compatibilidade de apresentação em diversos meios e tamanhos, ainda que pelas mesma linguagens (HTML, CSS, JS). Por exemplo, antes dos devices com telas, responsividade não era uma preocupação, passou a ser relevante que um projeto fosse responsivo gradualmente, de acordo com a popularização dos dispositivos dos usuários.

**Comment by rickbenetti** [comment]
Fui FrontEnd Engineer na Agência Click (hoje isobar) em 2008, e o termo já existia há muito tempo, acredito que deve ter surgido mesmo lá pra 2003 mais ou menos que foi quando a Web voltou a tomar fôlego no mercado internacional e aqui.

**Comment by kvnol** [comment]
[De acordo com o Google Trends](https://trends.google.com.br/trends/explore?cat=13&date=2004-01-01%202017-09-04&q=front-end%20developer), o termo "Front-end Developer" aparece em março de 2007. O jQuery foi lançado em dezembro de 2006, acho que tem muito a ver com o lançamento do jQuery mesmo, com o que o @vitocaio disse mesmo.

![image](https://user-images.githubusercontent.com/3299130/30043254-c0fc3b38-91cc-11e7-8c8e-75c79596b48f.png)

**Comment by bernardodiasc** [comment]
https://www.webdesignerdepot.com/2009/05/the-evolution-of-cell-phone-design-between-1983-2009/ front-end developer é o cara que desenvolve para o usuário, então é de fundamental relevancia o que o usuário utiliza para interagir com o software que esse profissional escreve.

**Comment by rubenmarcus** [comment]
gosto do Termo Webmaster kkkkkkkkkkkkkk mas Front-end também é bonito. Será que daqui 2 anos ainda estará na "moda" , ou será Fullstack a bola da vez?

**Reply by rodrigonahid** [reply]
Ainda está na moda kkkkk

**Comment by felipefialho** [comment]
Comecei a trabalhar na área em 2009, e foi só em 2011 que escutei o termo pela primeira vez.

Mas na real? 

Antes chamavam de Web Designer, que fazia mais ou menos as mesmas coisas (com um pouco mais de viés em Design), com o tempo o Front-Ender foi pegando tarefas mais complexas de integração (especialmente após o surgimento do NodeJS) e se afastando um pouco do antigo "Web Designer".

**Comment by vitocaio** [comment]
@rubenmarcus Webmaster era da hora mesmo! e acho que FullStack ja e a bola da vez, o cara que é front e não entendi nada de nodeJS ou algo do lado server, vai fica para trás

**Comment by andresantos-tech** [comment]
Isso depende de um monte de fatores. Em alguns contextos só fez sentido ter
um cara chamado de "frontend" quando surgiu a preocupação com mobile, em
outros esse profissional já era necessário quando começou a preocupação com
SEO, e por aí vai.

Se for colocar um ponto inicial, eu acho que a "separação" entre webmaster
e frontend no desenvolvimento web começou com a adoção do CSS2 e o conceito
de *tableless, *lá pra 2003/2004.

Depois disso, outros pontos importantes foram:

- Lá pra 2006, o surgimento de libs e frameworks pra trabalhar mais
facilmente com javascript (sobretudo jQuery).
- A morte do Flash, que começou em 2007 e se concretizou nos anos seguintes.
- A ação antitrust que a Microsoft sofreu por parte da União Européia (em
2008 ou 2009), que facilitou o uso de outros navegadores além do IE.
- Sem precisar se preocupar com versões antigas do IE, veio a evolução do
javascript que segue até hoje.

Com certeza deve ter mais coisa, mas os momentos principais que eu lembro
são esses.

Em 5 de setembro de 2017 11:52, vitocaio <notifications@github.com>
escreveu:

> @rubenmarcus <https://github.com/rubenmarcus> Webmaster era da hora
> mesmo! e acho que FullStack ja e a bola da vez, o cara que é front e não
> entendi nada de nodeJS ou algo do lado server, vai fica para trás
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/707#issuecomment-327200693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABNIbhhqM7K7eWyhcrxChYy_T5Oey0Heks5sfWAlgaJpZM4PMMy_>
> .
>



-- 
- André O. Santos -
